src/socket/send.rs

```diff
@@ -53,15 +72,23 @@ impl SendHandler {
     async fn start<P: ProtocolIdentity>(&mut self) {
         loop {
             tokio::select! {
-                Some(packet) = self.handler_recv.recv() => {
-                    let encoded_packet = packet.packet.encode::<P>(&packet.node_address.node_id);
-                    if encoded_packet.len() > MAX_PACKET_SIZE {
-                        warn!("Sending packet larger than max size: {} max: {}", encoded_packet.len(), MAX_PACKET_SIZE);
-                    }
-                    if let Err(e) = self.send.send_to(&encoded_packet, &packet.node_address.socket_addr).await {
+                Some(outbound) = self.handler_recv.recv() => {
+                    let (dst_addr, encoded_pkt) = match outbound {
+                        Outbound::Packet(outbound_packet) => {
+                            let dst_id = outbound_packet.node_address.node_id;
+                            let encoded_packet = outbound_packet.packet.encode::<P>(&dst_id);
+                            if encoded_packet.len() > MAX_PACKET_SIZE {
+                                warn!("Sending packet larger than max size: {} max: {}", encoded_packet.len(), MAX_PACKET_SIZE);
+                            }
+                            let dst_addr = outbound_packet.node_address.socket_addr;
+                            (dst_addr, encoded_packet)
+                        }
+                        Outbound::KeepHolePunched(dst) => (dst, vec![]),
+                    };
+                    if let Err(e) = self.send.send_to(&encoded_pkt, &dst_addr).await {
                         trace!("Could not send packet. Error: {:?}", e);
                     } else {
-                        METRICS.add_sent_bytes(encoded_packet.len());
+                        METRICS.add_sent_bytes(encoded_pkt.len());
```

Comment:
why are we changing ``encoded_packet`` to ``encoded_pkt`` this decreases readability with no received benefit? Could we revert this back to ``encoded_packet``